diff --git a/node_modules/@react-native-google-signin/google-signin/ios/RNGoogleSignin.mm b/node_modules/@react-native-google-signin/google-signin/ios/RNGoogleSignin.mm
index c56cac8..3730887 100644
--- a/node_modules/@react-native-google-signin/google-signin/ios/RNGoogleSignin.mm
+++ b/node_modules/@react-native-google-signin/google-signin/ios/RNGoogleSignin.mm
@@ -131,13 +131,20 @@ + (BOOL)requiresMainQueueSetup
 
       NSString* hint = options[@"loginHint"];
       NSArray* scopes = self.scopes;
+      NSString* nonce = options[@"nonce"];
 
 #if DEBUG
     @try {
 #endif
-      [GIDSignIn.sharedInstance signInWithPresentingViewController:presentingViewController hint:hint additionalScopes:scopes completion:^(GIDSignInResult * _Nullable signInResult, NSError * _Nullable error) {
-        [self handleCompletion:signInResult withError:error withResolver:resolve withRejector:reject fromCallsite:@"signIn"];
-      }];
+      if (nonce) {
+        [GIDSignIn.sharedInstance signInWithPresentingViewController:presentingViewController hint:hint additionalScopes:scopes nonce:nonce completion:^(GIDSignInResult * _Nullable signInResult, NSError * _Nullable error) {
+          [self handleCompletion:signInResult withError:error withResolver:resolve withRejector:reject fromCallsite:@"signIn"];
+        }];
+      } else {
+        [GIDSignIn.sharedInstance signInWithPresentingViewController:presentingViewController hint:hint additionalScopes:scopes completion:^(GIDSignInResult * _Nullable signInResult, NSError * _Nullable error) {
+          [self handleCompletion:signInResult withError:error withResolver:resolve withRejector:reject fromCallsite:@"signIn"];
+        }];
+      }
 #if DEBUG
     }
     @catch (NSException *exception) {
